### PR TITLE
gsdx windows: update gui to add a HPO v2 checkbox

### PR DESCRIPTION
**Pull request by gregory38**
Squash of PR #1758 

Need someone to check the result. I didn't understand why `IDC_OFFSETHACK_NEW` was added so I kept only `IDC_OFFSETHACK_TEXT` (that being said, it is maybe useless too)

### CODE REVIEW

File: plugins/GSdx/resource.h [hunk 1]
```diff
@@ -140,7 +141,7 @@
 #ifndef APSTUDIO_READONLY_SYMBOLS
 #define _APS_NEXT_RESOURCE_VALUE        10013
 #define _APS_NEXT_COMMAND_VALUE         32771
-#define _APS_NEXT_CONTROL_VALUE         2106
+#define _APS_NEXT_CONTROL_VALUE         2107
```
Comment (ssakash) [inline plugins/GSdx/resource.h]: Duplicate value? This macro should have a value of 2121.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 2]
```diff
@@ -723,7 +722,12 @@ void GSHacksDlg::OnInit()
 	CheckDlgButton(m_hWnd, IDC_FAST_TC_INV, theApp.GetConfigB("UserHacks_DisablePartialInvalidation"));
 	CheckDlgButton(m_hWnd, IDC_AUTO_FLUSH, theApp.GetConfigB("UserHacks_AutoFlush"));
 	CheckDlgButton(m_hWnd, IDC_UNSCALE_POINT_LINE, theApp.GetConfigB("UserHacks_unscale_point_line"));
-
+	std::vector<GSSetting> hpo_combobox = theApp.m_gs_offset_hack;
+	if (!ogl)
+	{
+		hpo_combobox.erase(hpo_combobox.begin() + 2, hpo_combobox.begin() + 4); //TODO: Enums would be more nicer than magic numbers.
+	}
+	ComboBoxInit(IDC_OFFSETHACK, hpo_combobox,theApp.GetConfigI("UserHacks_HalfPixelOffset");
```
Comment (ssakash) [inline plugins/GSdx/GSSettingsDlg.cpp]: Function missing closing paranthesis.

File: plugins/GSdx/GSSettingsDlg.cpp [hunk 3]
```diff
@@ -723,7 +722,12 @@ void GSHacksDlg::OnInit()
 	CheckDlgButton(m_hWnd, IDC_FAST_TC_INV, theApp.GetConfigB("UserHacks_DisablePartialInvalidation"));
 	CheckDlgButton(m_hWnd, IDC_AUTO_FLUSH, theApp.GetConfigB("UserHacks_AutoFlush"));
 	CheckDlgButton(m_hWnd, IDC_UNSCALE_POINT_LINE, theApp.GetConfigB("UserHacks_unscale_point_line"));
-
+	std::vector<GSSetting> hpo_combobox = theApp.m_gs_offset_hack;
+	if (!ogl)
+	{
+		hpo_combobox.erase(hpo_combobox.begin() + 2, hpo_combobox.begin() + 4); //TODO: Enums would be more nicer than magic numbers.
```
Comment (ssakash) [inline plugins/GSdx/GSSettingsDlg.cpp]: Well, best to remove the comment here if there are no plans to add enumerations.

**Comment by ssakash** [comment]
> Need someone to check the result. I didn't understand why IDC_OFFSETHACK_NEW was added so I kept only IDC_OFFSETHACK_TEXT (that being said, it is maybe useless too)

The define is needed, it's used to the gray out the respective GUI element when the user has selected a native resolution.

**Comment by ssakash** [comment]
By the way, since you've made a PR on your own, maybe you could also add the necessary Linux/INI variable changes in this PR and merge this in a single go?

**Comment by gregory38** [comment]
@ssakash I will add the linux bit later. First I want to have be sure woj work is correct. I updated based on your comment. I added back the `_NEW` define.

**Comment by ssakash** [comment]
> I updated based on your comment. I added back the _NEW define.

Ahh, I'm terribly sorry, I think I got it confused. That macro isn't needed, only the ``IDC_OFFSETHACK_TEXT`` define is needed. (``IDC_OFFSETHACK_TEXT`` is used for graying out the element)

**Comment by gregory38** [comment]
@ssakash you're lucky I didn't close my text editor :p I revert it. Just increase the final  value to 2121

**Comment by woj1993** [comment]
@gregory38 Thanks. Without help I probably would be still fixing it.

**Comment by ssakash** [comment]
> You need to delete line 751 or 757 because is it doubled. Everything else seems ok.

Nice catch, though make sure to do the redundant function removal in a way that ``EnableWindow()`` for ``IDC_OFFSETHACK_TEXT`` and ``IDC_OFFSETHACK`` are close together.

**Comment by FlatOutPS2** [comment]
Seems like it's done now. I've tested it and it looks good. Ready to merge?

**Comment by woj1993** [comment]
@FlatOutPS2 I don't find any errors.
P.S. What I need to do with my pull request?

**Comment by gregory38** [comment]
I added Linux change. It misses @ssakash approval.

@woj1993 I will close both PR. However you might need to recreate a fresh GIT for future contribution. I know it wasn't easy, thanks you for your perseverance.

File: plugins/GSdx/GSSetting.cpp [hunk 4]
```diff
@@ -55,7 +55,10 @@ const char* dialog_message(int ID, bool* updateText) {
 		case IDC_ALPHAHACK:
 			return "Different alpha handling. Can work around some shadow problems.";
 		case IDC_OFFSETHACK:
-			return "Might fix some misaligned fog, bloom, or blend effect.";
+			return "Might fix some misaligned fog, bloom, or blend effect.\n"
+				"The preferred option is Normal (Vertex) as it is most likely to resolve misalignment issues.\n"
+				"The special cases are only useful in couple of games like Captain Tsubasa.\n"
+				"\n Special cases are disabled from code side for now.";
```
Comment (turtleli) [inline plugins/GSdx/GSSetting.cpp]: Space between `\n` and Special should be removed.

Comment (FlatOutPS2) [inline plugins/GSdx/GSSetting.cpp]: The line can be removed instead. ;)

Comment (gregory38) [inline plugins/GSdx/GSSetting.cpp]: Ok latest commit ammended :+1:

File: plugins/GSdx/GSSetting.cpp [hunk 5]
```diff
@@ -55,7 +55,9 @@ const char* dialog_message(int ID, bool* updateText) {
 		case IDC_ALPHAHACK:
 			return "Different alpha handling. Can work around some shadow problems.";
 		case IDC_OFFSETHACK:
-			return "Might fix some misaligned fog, bloom, or blend effect.";
+			return "Might fix some misaligned fog, bloom, or blend effect.\n"
+				"The preferred option is Normal (Vertex) as it is most likely to resolve misalignment issues.\n"
+				"The special cases are only useful in couple of games like Captain Tsubasa.\n";
```
Comment (turtleli) [inline plugins/GSdx/GSSetting.cpp]: `\n` should be removed from the end of the line. Also `a` should be inserted before `couple`.

Comment (gregory38) [inline plugins/GSdx/GSSetting.cpp]: Done on first commit. Thank

**Comment by woj1993** [comment]
@gregory38 When Users hack HPO will have value 2 what will happen?

**Comment by ssakash** [comment]
> @gregory38 When Users hack HPO will have value 2 what will happen?

The false branch of ``if (UserHacks_HPO == 3)`` will be entered, that's where the necessary functional code for ``Special (Aggressive)`` is present.

**Comment by woj1993** [comment]
@ssakash Ok. Sorry I didn't view full files.

**Pull request merged**